dot-net-sdk/validators/RuleValidator.cs

```diff
@@ -2,115 +2,158 @@
 using eppo_sdk.dto;
 using static eppo_sdk.dto.OperatorType;
 using NuGet.Versioning;
+using eppo_sdk.helpers;
+using eppo_sdk.exception;
 
 namespace eppo_sdk.validators;
 
-public class RuleValidator
-{
-    public static Rule? FindMatchingRule(SubjectAttributes subjectAttributes, List<Rule> rules)
-    {
-        return rules.Find(rule => MatchesRule(subjectAttributes, rule));
-    }
 
-    private static bool MatchesRule(SubjectAttributes subjectAttributes, Rule rule)
-    {
-        List<bool> conditionEvaluations = EvaluateRuleCondition(subjectAttributes, rule.conditions);
-        return !conditionEvaluations.Contains(false);
-    }
 
-    private static List<bool> EvaluateRuleCondition(SubjectAttributes subjectAttributes, List<Condition> ruleConditions)
+public static partial class RuleValidator
+{
+    public static FlagEvaluation? EvaluateFlag(Flag flag, string subjectKey, Subject subjectAttributes)
     {
-        return
-            ruleConditions.ConvertAll(condition => EvaluateCondition(subjectAttributes, condition));
-    }
+        if (!flag.enabled) return null;
 
-    private static bool EvaluateCondition(SubjectAttributes subjectAttributes, Condition condition)
-    {
-        try
+        var now = DateTimeOffset.Now.ToUniversalTime();
+        foreach (var allocation in flag.allocations)
         {
-            // Operators other than `IS_NULL` need to assume non-null
-            if (condition.operatorType == IS_NULL) {
-                bool isNull = !subjectAttributes.TryGetValue(condition.attribute, out Object? outVal) || HasEppoValue.IsNullValue(new HasEppoValue(outVal));
-                return condition.BoolValue() == isNull;
-            }
-            else if (subjectAttributes.TryGetValue(condition.attribute, out Object? outVal))
+            if (allocation.startAt.HasValue && allocation.startAt.Value > now || allocation.endAt.HasValue && allocation.endAt.Value < now)
             {
-                var value = new HasEppoValue(outVal!); // Assuming non-null for simplicity, handle nulls as necessary
+                continue;
+            }
 
-                if (condition.operatorType == GTE)
+            subjectAttributes["id"] = subjectKey;
```

Comment:
We should make sure we do not override the "id" field if it already exists in `subjectAttributes`

Reply:
done